src/main/java/hudson/plugins/warnings/tokens/AbstractDetailedTokenMacro.java

```diff
@@ -0,0 +1,132 @@
+package hudson.plugins.warnings.tokens;
+
+import hudson.plugins.analysis.core.BuildResult;
+import hudson.plugins.analysis.core.ResultAction;
+import hudson.plugins.analysis.tokens.AbstractTokenMacro;
+import hudson.plugins.analysis.util.model.FileAnnotation;
+import hudson.plugins.analysis.util.model.Priority;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Provides a token that evaluates detailed informations to the plug-in
+ * build result.
+ *
+ * @author Benedikt Spranger
+ */
+public abstract class AbstractDetailedTokenMacro extends AbstractTokenMacro {
+
+    public AbstractDetailedTokenMacro(String tokenName, Class<? extends ResultAction<? extends BuildResult>>... resultActions) {
+        super(tokenName, resultActions);
+    }
+
+    private final List<String> modules = new ArrayList<String>();
+    private boolean verbose = false;
+    private boolean showLow = true;
+    private boolean showNormal = true;
+    private boolean showHigh = true;
+    private String linechar = "-";
+
+    @Parameter
+    public int indent = 0;
+
+    @Parameter(alias="modules")
+    public void setModules(String module) {
+        modules.add(module);
+    }
+
+    @Parameter(alias="verbose")
+    public void setVerbosity(boolean verbosity) {
+        verbose = verbosity;
+    }
+
+    @Parameter(alias="low")
+    public void setLow(boolean show) {
+        showLow = show;
+    }
+
+    @Parameter(alias="normal")
+    public void setNormal(boolean show) {
+        showNormal = show;
+    }
+
+    @Parameter(alias="high")
+    public void setHigh(boolean show) {
+        showHigh = show;
+    }
+
+    @Parameter(alias="linechar")
+    public void setLineChar(String c) {
+        linechar = c.substring(0, 1);
+    }
+
+    protected String evalWarnings(final BuildResult result, Collection<FileAnnotation> warnings) {
+        String messages = "";
+
+        if (modules.isEmpty())
+            modules.add("all");
+
+        for (String module : modules) {
+            boolean allWarn = module.equals("all");
+            String heading = module + " annotations:";
+            String tmp = "";
+
+            for (FileAnnotation annotation : warnings) {
+                Priority prio = annotation.getPriority();
+                if (prio == Priority.LOW && !showLow)
```

Comment:
Can you please reformat and enforce that always braces are used (see [coding style](https://github.com/uhafner/codingstyle)).